### ISSUE TITLE
Can't get series list from theme DAO

### ISSUE THREAD

**Issue by xhqiao89** [xhqiao89/HydroDesktop_test#66]
<b>twhitvine[CodePlex]</b> <br />Here's my code, followed by the error. The code brings back the Theme object with the right theme name, but I can't get a list of series for the theme. What's wrong?

int themeID = Convert.ToInt32(cbVariable.SelectedValue);
HydroDesktop.Database.RepositoryManager repositoryManager = HydroDesktop.Database.Config.RepositoryManager;
repositoryManager.BeginTransaction ();
HydroDesktop.ObjectModel.Theme theme = repositoryManager.ThemeDAO.FindUnique ( quotIdquot, themeID );
System.Collections.Generic.IListltHydroDesktop.ObjectModel.Seriesgt seriesList = theme.SeriesList;
// Error happens on the next line
foreach ( HydroDesktop.ObjectModel.Series series in seriesList ) 
{
System.Windows.Forms.MessageBox.Show ( series.ValueCount.ToString () );
}
repositoryManager.Commit ();
and the error:

NHibernate.LazyInitializationException was unhandled
Message=quotInitializing[HydroDesktop.ObjectModel.Theme#8]-failed to lazily initialize a collection of role: HydroDesktop.ObjectModel.Theme.SeriesList, no session or session was closedquot

**Comment by xhqiao89** [comment]
<b>valentinedwv[CodePlex]</b> <br />Most fetching is lazy fetching, which means objects are retrieved as needed.

Stateless session means that the object is disconnected from the database, so if the object has not be retrieved, then there is no data connection to retrieve it.

**Comment by xhqiao89** [comment]
<b>jirikadlec2[CodePlex]</b> <br />I have committed a fix in the code. I have also created a unit test DatabaseTests/DataManagerTests/ThemeDAOTest.cs to verify that this item is fixed. The reason for the error was the use of NHibernate 'StatelessSession'. The 'StatelessSession seems to
 be faster for write operations but it doesn't support accessing related objects such as theme.SeriesList or series.Variable.TimeUnit.
